src/lib/dai.c

```diff
@@ -123,6 +128,86 @@ void dai_group_put(struct dai_group *group)
 		group->group_id = 0;
 }
 
+#if CONFIG_ZEPHYR_NATIVE_DRIVERS
+/* called from ipc/ipc3/handler.c and some platform.c files */
+struct dai *dai_get(uint32_t type, uint32_t index, uint32_t flags)
+{
+	char dmic_name[] = "DMICX";
+	char ssp_name[] = "SSP_X";
+	char alh_name[] = "ALH_X";
+	const struct device *drv;
+	char *drv_name = NULL;
+	struct dai *d;
+
+	tr_info(&dai_comp_tr, "get_dai_zephyr index: %u type: %u", index, type);
+
+	switch (type) {
+	case SOF_DAI_INTEL_SSP:
+		drv_name = ssp_name;
+		break;
+	case SOF_DAI_INTEL_ALH:
+		drv_name = alh_name;
+		break;
+	case SOF_DAI_INTEL_DMIC:
+		drv_name = dmic_name;
+		break;
+	default:
+		break;
+	}
+
+	if (!drv_name)
+		return NULL;
+
+	drv_name[4] = (uint8_t)((index % 10UL) + '0');
```

Comment:
minor: I think this could be a good actual usage of an assert to check that strlen of drv_name is sufficient.

Reply:
oh, something is missing as I changed this to Liams suggerstion.... hold on for awhile

Reply:
@kv2019i I screwed up yesterday with git reset... now includes the "dai array" changes suggested by Liam for this part of the code... apologies